math-scala/src/main/scala/org/apache/mahout/math/drm/logical/CheckpointAction.scala

```diff
@@ -44,5 +44,6 @@ abstract class CheckpointAction[K: ClassTag] extends DrmLike[K] {
     case Some(cp) => cp
   }
 
+  val classTag = implicitly[ClassTag[K]]
```

Comment:
yes you can't put implementation into trait. we'd have to make it optional
for the trait (=???) and override in the concrete things. Concrete things
are of course just of two types: logical operators based on the classes i
already mentioned, that already have this support, and the checkpoint
implementation, which probably does not yet. let me play with your branch a
little to see what we can do.

On Tue, Sep 29, 2015 at 11:10 AM, Alexey Grigorev <notifications@github.com>
wrote:

> In
> math-scala/src/main/scala/org/apache/mahout/math/drm/logical/CheckpointAction.scala
> <https://github.com/apache/mahout/pull/137#discussion_r40706926>:
>
> > @@ -44,5 +44,6 @@ abstract class CheckpointAction[K: ClassTag] extends DrmLike[K] {
> >      case Some(cp) => cp
> >    }
> >
> > +  val classTag = implicitly[ClassTag[K]]
>
> I remember trying to attach it to DrmLike, but I had some problems because
> it's a trait
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/mahout/pull/137/files#r40706926>.
>